File: app/src/main/java/com/nolamarel/onlinelibrary/Adapters/sections/SectionAdapter.java
package com.nolamarel.onlinelibrary.Adapters.sections;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.nolamarel.onlinelibrary.OnItemClickListener;
import com.nolamarel.onlinelibrary.R;

import java.util.ArrayList;

public class SectionAdapter extends RecyclerView.Adapter<SectionAdapter.SectionViewHolder> {
private ArrayList<Section> sections = new ArrayList<>();
    private OnItemClickListener.ItemClickListener listener;

    public SectionAdapter(ArrayList<Section> sections, OnItemClickListener.ItemClickListener listener) {
        this.listener = (OnItemClickListener.ItemClickListener) listener;
        this.sections = sections;
    }

public SectionAdapter(ArrayList<Section> sections){
    this.sections = sections;
}

    @NonNull
    @Override
    public SectionViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.sections_item, parent, false);
        return new SectionViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull SectionViewHolder holder, int position) {
        Section section = sections.get(position);

        holder.section_name.setText(section.sectionName);
        Glide.with(holder.itemView.getContext()).load(section.sectionIv).into(holder.section_iv);

        holder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                    int clickedPosition = holder.getAdapterPosition();
                    if (clickedPosition != RecyclerView.NO_POSITION) {
                        listener.onItemClick(clickedPosition);
                    }

            }
        });
    }

    @Override
    public int getItemCount() {
        return sections.size();
    }

    public class SectionViewHolder extends RecyclerView.ViewHolder{

    TextView section_name;
    ImageView section_iv;

        public SectionViewHolder(@NonNull View itemView) {
            super(itemView);

            section_name = itemView.findViewById(R.id.section_name);
            section_iv = itemView.findViewById(R.id.section_iv);
        }
    }

}

File: app/src/main/java/com/nolamarel/onlinelibrary/Adapters/sections/SectionAdapter.kt
package com.nolamarel.onlinelibrary.Adapters.sections

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.nolamarel.onlinelibrary.OnItemClickListener.ItemClickListener
import com.nolamarel.onlinelibrary.R

class SectionAdapter(
    private var sections: ArrayList<Section>,
    private var listener: ItemClickListener? = null
) : RecyclerView.Adapter<SectionAdapter.SectionViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): SectionViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.sections_item, parent, false)
        return SectionViewHolder(view)
    }

    override fun onBindViewHolder(holder: SectionViewHolder, position: Int) {
        val section = sections[position]

        holder.sectionName.text = section.sectionName

        if (!section.sectionIv.isNullOrBlank()) {
            Glide.with(holder.itemView.context)
                .load(section.sectionIv)
                .placeholder(R.drawable.books)
                .error(R.drawable.books)
                .into(holder.sectionImage)
        } else {
            holder.sectionImage.setImageResource(R.drawable.books)
        }

        holder.itemView.setOnClickListener {
            val clickedPosition = holder.adapterPosition
            if (clickedPosition != RecyclerView.NO_POSITION) {
                listener?.onItemClick(clickedPosition)
            }
        }
    }

    override fun getItemCount(): Int = sections.size

    inner class SectionViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val sectionName: TextView = itemView.findViewById(R.id.section_name)
        val sectionImage: ImageView = itemView.findViewById(R.id.section_iv)
    }
}